src/util/http/websocket/QueryToSocketDistributor.cpp

```diff
@@ -12,19 +12,20 @@
 
 namespace ad_utility::websocket {
 
+auto QueryToSocketDistributor::useStrandedAwaitable() const {
+  return net::bind_executor(strand_, net::use_awaitable);
+}
+
 net::awaitable<void> QueryToSocketDistributor::waitForUpdate() const {
-  try {
-    co_await infiniteTimer_.async_wait(net::use_awaitable);
-    AD_THROW("Infinite timer expired. This should not happen");
-  } catch (boost::system::system_error& error) {
-    if (error.code() != net::error::operation_aborted) {
-      throw;
-    }
-  }
+  auto [error] =
+      co_await infiniteTimer_.async_wait(net::as_tuple(useStrandedAwaitable()));
```

Comment:
Out of interest:
doe `as_tuple(use_awaitable)` do something smarter than `try {...} catch{...}`?